app/controllers/pets_controller.rb

```diff
@@ -65,6 +75,26 @@ def destroy_from_shelter
   end
 
   private
+  def empty_fields(current_params)
+    @pet_empty_fields = []
+    current_params.each do |key, value|
+      if value == ""
+        @pet_empty_fields << key.capitalize
+      end
+    end
+    @pet_empty_fields
+    # This looks like a RUBY .reduce opportunity for creating that new array
+  end
+
+  def empty_fields_convert
```

Comment:
I like the refactoring that we did here.

Reply:
Woo! I don't know now if we can take these to application controller, but if there is time, we can work on refactoring these methods into app_cont :)